Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2
# Dark Portal Mechanics

## Developer changelog:
- Added the **_Dark Portal_** building with 2 stages: the **_Dark Portal_** and the **_Dark Rift_**. The Rift has very small chances of Draenor event happening and gives no bonuses.
- The **_Dark Portal_** can be destroyed becoming the **_Dark Rift_** either by a siege or by a decision if you control the Dark Portal county.
- Renamed the Dark Portal county "Dark Mire".
- The **_Invite Remaining Clans_** decision of the Warchief requires the Dark Portal to exist.
- Winning the Horde gives you the Dark Portal county and lands around to destroy the **_Dark Portal_** if you can.
- Random events:
  - **_Draenor Refugees Event_**:
    - May happen for rulers in the Black Morass, around the Dark Portal.
    - Spawns a draenei, orc, ogre, or arakkoa and lets you decide their fate: keep, kick or imprison.
    - `event WCDRA.100` in the console to force the event.

# Tasks:
- [x] Add a notification event
- [x] Block Inviting Clans if the Dark Portal is closed
- [x] Move `GetDarkPortalRift` to this branch
- [x] Move event localization to separate file
- [x] Test vrykul adventurers
- [x] Stop spawn vrykul if they can't conquer anything
- [x] Find out why orcs can't declare wars. Maybe there's big cultural presence in region?
- [x] Random events:
  - [x] Refugees try to pass the Portal (draenei, arakkoa, ogres, orcs)
- [x] Take Portal in CBs
- [x] Use different values for stress, with less stress impact


# How to test:
@Warcraft-GoA-Development-Team/testers test the mentioned in the changelog features and tell your opinion.

@Warcraft-GoA-Development-Team/english-proofreaders open the [Page](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/395/files?file-filters%5B%5D=.yml), scroll to the bottom to the `.yml` files, check them, suggest changes. If you forgot, here's the [Suggestion Guidelines](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/wiki/How-to-be-Tester-(For-CK3)#reviewing-code).
